pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -821,15 +834,53 @@ func (a *Operator) copyToNamespace(prototype *v1alpha1.ClusterServiceVersion, ns
 	} else if err != nil {
 		return nil, err
 	}
+	// metadata drift guard: detect manual modifications to spec or status
+	if og, orv := existing.Annotations[observedGenerationAnnotation], existing.Annotations[observedResourceVersionAnnotation]; (og != "" && og != fmt.Sprint(existing.GetGeneration())) || (orv != "" && orv != existing.ResourceVersion) {
+		// full resync for metadata drift
+		// prepare prototype for update
+		prototype.Namespace = existing.Namespace
+		prototype.ResourceVersion = existing.ResourceVersion
+		prototype.UID = existing.UID
+		// sync hash annotations
+		prototype.Annotations[nonStatusCopyHashAnnotation] = nonstatus
+		prototype.Annotations[statusCopyHashAnnotation] = status
```

Comment:
should we update the status hash post status update, i.e. with the observed resource version and generation annotations (for the same reason described in line 897)?

Reply:
done in soon to be pushed commit 4b21aa6e